crates/keystore/src/errors.rs

```diff
@@ -0,0 +1,23 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum KeystoreError {
+    #[error("File error: {0}")]
+    FileError(std::io::Error),
+    #[error("JwkConversionError")] 
+    JwkConversionError,
+    #[error("KeyPairGenerationError")]
+    KeyPairGenerationError,
+    #[error("non compliant")]
+    NonCompliant,
+    #[error("not found")]
+    NotFound,
+    #[error("parse error")]
+    ParseError(serde_json::Error),
+    #[error("serde error")]
+    SerdeError(serde_json::Error),
+    #[error("Encryption error: {0}")]
+    EncryptionError(chacha20poly1305::Error),
+    #[error("Decryption error: {0}")]
+    DecryptionError(chacha20poly1305::Error),
+}
```

Comment:
Your errors need to be more descriptive:
```suggestion
use thiserror::Error;

#[derive(Debug, Error)]
pub enum KeystoreError {
    #[error("File operation failed: {0}")]
    FileError(#[from] std::io::Error),

    #[error("JWK conversion failed")]
    JwkConversionError,

    #[error("Key pair generation failed")]
    KeyPairGenerationError,

    #[error("Non-compliant data")]
    NonCompliant,

    #[error("Item not found")]
    NotFound,

    #[error("Failed to parse JSON data: {0}")]
    ParseError(#[from] serde_json::Error),

    #[error("Serialization error: {0}")]
    SerializationError(#[from] serde_json::Error),
    
    #[error("Deserialization error: {0}")]
    DeserializationError(#[from] serde_json::Error),

    #[error("Encryption failed: {0}")]
    EncryptionError(#[from] chacha20poly1305::Error),

    #[error("Decryption failed: {0}")]
    DecryptionError(#[from] chacha20poly1305::Error),
}
```